src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs

```diff
@@ -103,18 +103,16 @@ internal AveragedPerceptronTrainer(IHostEnvironment env, Arguments args)
         /// <param name="decreaseLearningRate">Wheather to decrease learning rate as iterations progress.</param>
         /// <param name="l2RegularizerWeight">L2 Regularization Weight.</param>
         /// <param name="numIterations">The number of training iteraitons.</param>
-        /// <param name="advancedSettings">A delegate to supply more advanced arguments to the algorithm.</param>
-        public AveragedPerceptronTrainer(IHostEnvironment env,
+        internal AveragedPerceptronTrainer(IHostEnvironment env,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
             string weights = null,
```

Comment:
delete..

Reply:
I do see it being used to initialize `InitialWeights`.   Thoughts ?

---
In reply to: [248544047](https://github.com/dotnet/machinelearning/pull/2163#discussion_r248544047) [](ancestors = 248544047)

Reply:
I meant the constructor, bad selection. 

---
In reply to: [248768887](https://github.com/dotnet/machinelearning/pull/2163#discussion_r248768887) [](ancestors = 248768887,248544047)

Reply:
Deleting the constructor u mean ?  That would be #2100

---
In reply to: [248834137](https://github.com/dotnet/machinelearning/pull/2163#discussion_r248834137) [](ancestors = 248834137,248768887,248544047)